docs/src/main/asciidoc/tls-registry-reference.adoc

```diff
@@ -883,3 +883,154 @@ On Mac, you can use the following command:
+----
+quarkus tls-registry lets-encrypt prepare --domain=<domain-dns-name>
+----
+
+[IMPORTANT]
+----
+Make sure you run a prepare command in the root directory of your application.
+----
+
+The `prepare` command does the following:
+
+- Creates a `.letsencrypt` folder in your application's root directory
+- Creates a self-signed domain certificate and private key for your application configured in the previous <<lets-encrypt-prerequisites>> step to be able to start and accept HTTPS requests.
+
+=== Start your application
+
+You can start your aplication in devmode:
+
+include::{includes}/devtools/dev.adoc[]
+
+Access your application endpoint using `https://localhost:8443/`, for example, `https://localhost:8443/hello`, accept a self-signed domain certificate in the browser.
+
+Next, request your first Let's Encrypt certificate.
+
+[[lets-encrypt-issue-certificate]]
+=== Issue certificate
+
+Run the `issue-certificate` command to acquire your first Let's Encrypt certificate:
+
+[source, shell]
+----
+quarkus tls-registry lets-encrypt issue-certificate
+--domain=<domain-dns-name> <1>
+--email=<your contact email> <2>
+--management-url=https://localhost:9000 <3>
+----
+<1> Set your domain name.
+<2> Provide your contact email address that Let's Encrypt can use to contact you in case of any issues with your Let's Encrypt account.
+<3> Set your application management URL which can be used to handle ACME challenges. Use `https://localhost:8443/` if you chose not to enable a management router in the <<lets-encrypt-prerequisites>> step.
+
+During this command, TLS registry CLI checks a Quarkus management router's readiness, creates and records a Let's Encrypt account information, issues a Let's Encrypt certificate request, and communicates with a Quarkus management router to have ACME challenges resolved.
+
+Once the Let's Encrypt certificate chain and private key have been successfully acquired, they are converted to PEM format and copied to your application's `.letsencrypt` folder. TLS registry is informed that a new certificate and private key are ready and it reloads them automatically.
+
+Now, access your application's endpoint using `https://localhost:8443/` again. Confirm in the browser that your domain certificate is now signed by the Let's Encrypt certificate authority.
+
+Note that currently, a Let's Encrypt account is created implicitly by the `issue-certificate` command to make it easy for users to get started with the ACME protocol. Support for the Let's Encrypt account management will evolve further.
+
+[[lets-encrypt-renew-certificate]]
+=== Renew certificate
+
+Renewing certificates is similar to issuing the first certificate, but it requires an existing account created during the <<lets-encrypt-issue-certificate>> step.
+
+Run the following command to renew your Let's Encrypt certificate:
+
+[source, shell]
+----
+quarkus tls-registry lets-encrypt renew-certificate
+--domain=<domain-dns-name> <1>
+----
+<1> Set your domain name.
```

Comment:
I'm wondering if we should not sotre this in the .letsencrypt directory actually (as an improvement for later), WDYT?